containers/go-postgres/README.md

```diff
@@ -0,0 +1,146 @@
+2. [Tips on populating data](https://www.postgresql.org/docs/14/populate.html) in the database. 
+
+If needed, you can use `postCreateCommand` to run commands after the container is created, by updating `.devcontainer/devcontainer.json` similar to what follows:
+
+```json
+"postCreateCommand": "go version && git --version && node --version"
+```
+
+### Adding another service
+
+You can add other services to your `docker-compose.yml` file [as described in Docker's documentation](https://docs.docker.com/compose/compose-file/#service-configuration-reference). However, if you want anything running in this service to be available in the container on localhost, or want to forward the service locally, be sure to add this line to the service config:
+
+```yaml
+# Runs the service on the same network as the database container, allows "forwardPorts" in devcontainer.json function.
+network_mode: service:db
+```
+
+### Installing GO Dependencies
+
+This definition includes the popular [PostGres Driver Library for Go](github.com/lib/pq). This is the recommended driver for use with Go, as per [GoLang Documentation](https://golangdocs.com/golang-postgresql-example).
+
+If you wish to change this, you may add additional `RUN` commands in the [Go Dockerfile](.devcontainer/Dockerfile). For example:
+
+```yaml
+# This line can be modified to add any needed additional packages
+RUN go get -x <github-link-for-package>
+```
+
+## Testing the definition
+
+This definition includes some test code that will help you verify it is working as expected on your system. Follow these steps:
+
+1. If this is your first time using a development container, please follow the [getting started steps](https://aka.ms/vscode-remote/containers/getting-started) to set up your machine.
+2. Clone this repository.
+3. Start VS Code, press <kbd>F1</kbd>, and select **Remote-Containers: Open Folder in Container...**
+4. Select the `containers/go` folder.
+5. After the folder has opened in the container, press <kbd>F5</kbd> to start the project.
+6. From here the following should print out in the Debug Console:
+   ```
+   Connected!
+   Sending Query to Database
+   One database in this cluster is: postgres
+   ```
+7. From here, you can add breakpoints or edit the contents of the `test-project` folder to do further testing.
+8. The Application can also be tested by running `./test.sh` from the Terminal. This will test the `hello.go` application.
+
+### Debugging Security
+To allow C++ debuggers to run within the Docker Containers, the [docker-compose.yml](.devcontainer/docker-compose.yml) contains the following lines:
+
+```yaml
+    security_opt:
+      - seccomp:unconfined
+    cap_add:
+      - SYS_PTRACE
+```
+
+As these can create security vulnerabilities, it is advisable to not use this unless needed. This should only be used in a Debug or Dev container, not in Production.
```

Comment:
Should these lines be commented out in the yaml file, and the user can decide if they want to uncomment them for use? Or is there a reason they are included?

Reply:
We should follow the Secure By Default principle whereever possible. The test app runs in the container from the command line and under the debugger even when running under the lower privilege `vscode` user account. 

Let's pick the secure default and document exceptions that need to lower security settings.

Reply:
Done.